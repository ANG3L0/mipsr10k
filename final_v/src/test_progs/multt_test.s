
        data1 = 0x0001   
        data2 = 0x000a
        data3 = 0x0002
        data4 = 0x0008

        lda  $r1, data1   /* $r1 = 1 */
        lda  $r2, data2   /* $r2 = 2 */
        lda  $r3, data3   /* $r3 = 6 */
        lda  $r4, data4   /* $r4 = 8 */
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
        mulq $r3, $r3, $r3
#       mult $r, $r1, $r2
        call_pal 0x555
