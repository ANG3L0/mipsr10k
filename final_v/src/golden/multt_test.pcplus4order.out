4:lda
8:lda
12:lda
16:lda
20:mulq
24:mulq
28:mulq
32:mulq
36:mulq
40:mulq
44:mulq
48:mulq
52:mulq
56:mulq
60:mulq
64:mulq
68:mulq
72:mulq
76:mulq
80:mulq
84:mulq
88:mulq
92:mulq
96:mulq
100:mulq
104:mulq
108:mulq
112:mulq
116:mulq
120:mulq
124:mulq
128:mulq
132:mulq
136:mulq
140:mulq
144:mulq
148:mulq
152:mulq
156:mulq
160:mulq
164:mulq
168:mulq
172:halt
